### PR TITLE
WD FT JUN 2022 | Ruben Andino

### PR DESCRIPTION
**Pull request by rubenanlo**
Exercise done except the last bonus iteration. For some reason my code doesn't allow me to do a push to a new array. I believe it's because the array I am trying to pull information from is actually an object.

### CODE REVIEW

File: src/movies.js [hunk 1]
```diff
@@ -1,28 +1,156 @@
...
+  
+  const sortedFullArr = titleArray.sort();
+  
+  if (sortedFullArr.length < 20) {
+    number = sortedFullArr.length;
+  } else {
+    number = 20;
+  }
+  
+  sortedArr = sortedFullArr.splice(0, number);
```
Comment (TA-Remote) [inline src/movies.js:100]: This is quite bad practice - make sure to always use `const` or `let` when declaring a variable

**Review by TA-Remote — COMMENTED** [review]
Great job here! Added some little notes to improve code.

regarding iteration 8 -> from what I can see so far, you are never actually checking the score of each movie... What the task is actually asking. 

- you got array of unique years 👌 
- next you need to get the average movie score of every year 
- compare which year has the highest average

Any questions/doubt - let me know :)
Karina

File: src/movies.js [hunk 2]
```diff
@@ -1,28 +1,156 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+  moviesArray = moviesArray.map((movie) => movie.director);
```
[the remark this answers is not in the document]
Comment (rubenanlo) [reply src/movies.js:5]: Thanks! It makes totally sense

File: src/movies.js (hunk 1 above)
Comment (rubenanlo) [reply src/movies.js:100]: I feel like I need to pracitce scoping :)

File: src/movies.js [hunk 3]
```diff
@@ -1,28 +1,156 @@
...
+    return 0;
+  }
+  
+  const movStevenArr = moviesArray.filter(
+    (movie) => movie.director === 'Steven Spielberg'
+  );  
+
+  const result = movStevenArr.filter(
+    (movie) => movie.genre.indexOf(`Drama`) >= 0
+  );
```
[the remark this answers is not in the document]
Comment (rubenanlo) [reply src/movies.js:29]: Thanks!

**Pull request closed without merging**